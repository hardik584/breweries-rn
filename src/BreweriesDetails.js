import React, {useEffect, useState} from 'react';
import {
  ActivityIndicator,
  FlatList,
  Text,
  View,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  Linking,
} from 'react-native';
import CustomStyle from './CustomStyle';

export default BreweriesDetails = props => {
  const {breweriesItem} = props.route.params;
  return (
    <SafeAreaView style={{flex: 1}}>
      <ScrollView>
        {
          <View>
            {Object.keys(breweriesItem).map((key, index) => {
              return (
                <View
                  key={key}
                  style={[
                    CustomStyle.detailsBackgroundStyle,
                    {backgroundColor: index % 2 == 0 ? '#ddd' : 'white'},
                  ]}>
                  <Text style={CustomStyle.baseText}>
                    {key.charAt(0).toUpperCase() + key.slice(1)}:
                    <Text style={    CustomStyle.innerText}>
                      {} {breweriesItem[key] ?? '-'}{' '}
                    </Text>
                  </Text>
                </View>
              );
            })}
          </View>
        }
      </ScrollView>
    </SafeAreaView>
  );
};
