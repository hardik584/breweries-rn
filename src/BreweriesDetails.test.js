import BreweriesDetails from './BreweriesDetails';
import renderer from 'react-test-renderer';
import * as React from 'react';

it(`Checking Breweries Details Rendering`, () => {
  let data = {"id":"alphabet-city-brewing-co-new-york","name":"Alphabet City Brewing Co","brewery_type":"contract","street":"96 Avenue C Frnt 4","address_2":null,"address_3":null,"city":"New York","state":"New York","county_province":null,"postal_code":"10009-7055","country":"United States","longitude":null,"latitude":null,"phone":"9143566809","website_url":"http://www.acbnyc.com","updated_at":"2022-08-20T02:56:08.975Z","created_at":"2022-08-20T02:56:08.975Z"};
  // const tree = renderer.create(<BreweriesDetails route={{"params":{"breweriesItem":data}}}> </BreweriesDetails>);
  const tree = renderer.create(<BreweriesDetails route={{"params":{"breweriesItem":data}}}> </BreweriesDetails>);

  expect(tree).toMatchSnapshot();
});
