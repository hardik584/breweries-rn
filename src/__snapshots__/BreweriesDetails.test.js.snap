// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Breweries Details Rendering 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Id
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              alphabet-city-brewing-co-new-york
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Name
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              Alphabet City Brewing Co
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Brewery_type
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              contract
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Street
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              96 Avenue C Frnt 4
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Address_2
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              -
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Address_3
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              -
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            City
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              New York
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            State
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              New York
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            County_province
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              -
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Postal_code
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              10009-7055
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Country
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              United States
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Longitude
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              -
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Latitude
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              -
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Phone
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              9143566809
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Website_url
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              http://www.acbnyc.com
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Updated_at
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              2022-08-20T02:56:08.975Z
               
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Created_at
            :
            <Text
              style={
                Object {
                  "color": "grey",
                  "fontSize": 15,
                }
              }
            >
               
              2022-08-20T02:56:08.975Z
               
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
